tests/pyfunc/test_spark.py

```diff
@@ -309,6 +309,95 @@ def predict(self, context, model_input):
         assert result.r7.tolist() == ["abc"] * 2
 
 
+def test_spark_udf_colspec_struct_return_type_inferring(spark):
+    class TestModel(PythonModel):
+        def predict(self, context, model_input):
+            input_len = len(model_input)
+            return {
+                "r1": [1] * input_len,
+                "r2": [1.5] * input_len,
+                "r3": [True] * input_len,
+                "r4": ["abc"] * input_len,
+            }
+
+    with mlflow.start_run() as run:
+        mlflow.pyfunc.log_model("model", python_model=TestModel(), signature=ModelSignature(
+            inputs=Schema([ColSpec("long")]),
+            outputs=Schema([
+                ColSpec('integer', 'r1'),
+                ColSpec('float', 'r2'),
+                ColSpec('boolean', 'r3'),
+                ColSpec('string', 'r4'),
+            ])
+        ))
+
+        udf = mlflow.pyfunc.spark_udf(
+            spark,
+            "runs:/{}/model".format(run.info.run_id),
+        )
+
+        data1 = spark.range(2).repartition(1)
+        result = (
+            data1.withColumn("res", udf("id"))
+            .select("res.r1", "res.r2", "res.r3", "res.r4")
+            .toPandas()
+        )
+        assert result.r1.tolist() == [1] * 2
+        np.testing.assert_almost_equal(result.r2.tolist(), [1.5] * 2)
+        assert result.r3.tolist() == [True] * 2
+        assert result.r4.tolist() == ["abc"] * 2
```

Comment:
/ai `result` in the following code is a pandas dataframe. Rewrite this using subscript (e.g. `result["a"]`). Please explain why we should use subscript.